src/app/parcels/ParcelsPage.tsx

```diff
@@ -522,139 +339,98 @@ const ParcelsPage: React.FC<{}> = () => {
         })();
     }, []);
 
-    useEffect(() => {
-        if (!areFiltersLoadingForFirstTime) {
-            const allFilters = [...primaryFilters, ...additionalFilters];
-            setIsLoading(true);
-            if (parcelsTableFetchAbortController.current) {
-                parcelsTableFetchAbortController.current.abort("stale request");
-            }
+    const fetchAndDisplayParcelsData = useCallback(async (): Promise<void> => {
+        const allFilters = [...primaryFilters, ...additionalFilters];
 
-            parcelsTableFetchAbortController.current = new AbortController();
+        if (parcelsTableFetchAbortController.current) {
+            parcelsTableFetchAbortController.current.abort("stale request");
+        }
+
+        parcelsTableFetchAbortController.current = new AbortController();
+
+        if (parcelsTableFetchAbortController.current) {
+            setErrorMessage(null);
+            setIsLoading(true);
 
-            if (parcelsTableFetchAbortController.current) {
-                setErrorMessage(null);
-                getParcelsDataAndCount(
+            try {
+                const { data, count } = await getParcelsDataAndCount(
                     supabase,
                     allFilters,
                     sortState,
                     parcelsTableFetchAbortController.current.signal,
                     startPoint,
                     endPoint
-                )
-                    .then(({ data, count }) => {
-                        setParcelsDataPortion(data);
-                        setFilteredParcelCount(count);
-                    })
-                    .catch((error) => {
-                        if (error instanceof DatabaseError) {
-                            setErrorMessage(error.message);
-                        }
-                    })
-                    .finally(() => {
-                        parcelsTableFetchAbortController.current = null;
-                        setIsLoading(false);
-                    });
+                );
+                setParcelsDataPortion(data);
+                setFilteredParcelCount(count);
+            } catch (error) {
+                if (error instanceof DatabaseError) {
+                    setErrorMessage(error.message);
+                }
+            } finally {
+                parcelsTableFetchAbortController.current = null;
+                setIsLoading(false);
             }
         }
-    }, [
-        startPoint,
-        endPoint,
-        primaryFilters,
-        additionalFilters,
-        sortState,
-        areFiltersLoadingForFirstTime,
-    ]);
+    }, [additionalFilters, endPoint, primaryFilters, sortState, startPoint]);
 
     useEffect(() => {
```

Comment:
Is seems like this original use effect is only subscribing to realtime updates. So I decided to do that at the beggining once rather than whenever the filters change.